src/js/menu/media-rendition-menu.ts

```diff
@@ -168,6 +172,10 @@ class MediaRenditionMenu extends MediaChromeMenu {
   }
 
   #onChange(): void {
+    const autoOption = this.defaultSlot.querySelector('[value="auto"]');
+    const isAuto = autoOption.part.contains('checked');
+    autoOption.textContent = isAuto ? `Auto (${this.mediaHeight}p)` : 'Auto';
```

Comment:
This works but it would be nice to keep the declarative "render" method for updating all state in the UI. Under this line `this.value = newValue ?? 'auto';` if you add a new line under it with `this.#render();` you get the same final result though all menu items will be rendered again.

My original intention was to optimize the render method later doing a simple diff of the menu items and updating only where needed. If you like you can try to add that.

High level an algorithm that compares the length of the menu items. Add or remove if needed.
I wouldn't bother with swapping and order, that's less likely an issue for our menus I think.
Once the length is correct, just compare the data with the DOM properties. If they don't match, set them (overwrite).